### ISSUE TITLE
`--cellranger_index` only appears to work if `--validate_params false` provided

### ISSUE THREAD

**Issue by nick-youngblut** [nf-core/scrnaseq#340]
### Description of the bug

The CellRanger index is a directory of files, and not just one. So, one should provide a directory via `--cellranger_index`; however, this throws the following error:

```
ERROR ~ ERROR: Validation of pipeline parameters failed!

 -- Check '.nextflow.log' file for details
The following invalid input values have been detected:

* --cellranger_index: '/large_experiments/multiomics/references/cellranger/refdata-gex-GRCh38-2024-A/' is not a file, but a directory (/large_experiments/multiomics/references/cellranger/refdata-gex-GRCh38-2024-A/)
```

It appears that the parameter validation is checking that the `--cellranger_index` is a file and not a directory. If I skip the validation via `--validate_params false`, then the pipeline will complete the CellRanger Count step. 

### Command used and terminal output

```console
nextflow run main.nf \
   -profile test_full,singularity \
   -process.executor slurm \
   -process.queue cpu_batch \
   --aligner cellranger \
   --cellranger_index /large_experiments/multiomics/references/cellranger/refdata-gex-GRCh38-2024-A/ \
   --outdir test_run_output \
   --validate_params false
```

### System information

Nextflow version: `23.10.1.5891`
Hardware: HPC
Executor: SLURM
Container engine: Apptainer
OS: Ubuntu
Version of nf-core/scrnaseq: 2.6.0

**Comment by nick-youngblut** [comment]
Along these same lines:

In `subworkflows/local/align_cellranger.nf`, the code is:

```
        assert cellranger_index || (fasta && gtf):
            "Must provide either a cellranger index or both a fasta file ('--fasta') and a gtf file ('--gtf')."

        if (!cellranger_index) {
            // Filter GTF based on gene biotypes passed in params.modules
            CELLRANGER_MKGTF( gtf )
            ch_versions = ch_versions.mix(CELLRANGER_MKGTF.out.versions)

            // Make reference genome
            CELLRANGER_MKREF( fasta, CELLRANGER_MKGTF.out.gtf, "cellranger_reference" )
            ch_versions = ch_versions.mix(CELLRANGER_MKREF.out.versions)
            cellranger_index = CELLRANGER_MKREF.out.reference
        }
```

...so based on "Must provide either a cellranger index or both a fasta file ('--fasta') and a gtf file ('--gtf').", one can assume that if `--cellranger_index` is provided, then `--gtf` and `--fasta` do not need to be provided.

However, if one does not not provide `--gtf` and `--fasta`, then warnings are generated:

```console
WARN: Access to undefined parameter `gtf` -- Initialise it to a default value eg. `params.gtf = some_value`
WARN: Access to undefined parameter `fasta` -- Initialise it to a default value eg. `params.fasta = some_value`
```

These warnings seem to be an unnecessary distraction (possibly leading to confusion), if the user provides `--cellranger_index` instead of  `--gtf` and `--fasta`.

**Comment by grst** [comment]
First problem should already be fixed in v2.7.
Second problem will be fixed in https://github.com/nf-core/scrnaseq/pull/354.

**Comment by grst** [comment]
Fixed in dev
